src/rust/logs-generator/src/main.rs

```diff
@@ -0,0 +1,436 @@
+            &wordlist,
+            min,
+            max,
+            known_words,
+            random_words,
+            random_digits,
+        ),
+        x => read_patterns(x.to_string()),
+    };
+
+    for pattern in &patterns {
+        print(verbose, format!("Pattern: {}", pattern));
+    }
+
+    let start = Instant::now();
+    let mut now = Instant::now();
+    let mut count_logs = 0;
+    let mut count_bytes = 0;
+    let mut current_logs = 0;
+    let mut current_bytes = 0;
+    let mut counter = 0;
+    loop {
+        // Print statistics
+        if now.elapsed().as_secs() >= time_resolution {
+            print(
+                verbose,
+                format!(
+                    "{} pps\t {} b/s",
+                    current_logs / now.elapsed().as_secs(),
+                    current_bytes / now.elapsed().as_secs()
+                ),
+            );
+            print(
+                verbose,
+                format!("Total stats: {} logs, {} bytes", count_logs, count_bytes),
+            );
+            now = Instant::now();
+            current_logs = 0;
+            current_bytes = 0;
+        }
+
+        // Skip iteration because limit of logs/s already reached
+        if logs_per_s > 0 && current_logs >= logs_per_s * time_resolution {
+            continue;
+        }
+
+        // Skip iteration because limit of bytes/s already reached
+        if bytes_per_s > 0 && current_bytes >= bytes_per_s * time_resolution {
+            continue;
+        }
+
+        // Stop generation if configured duration reached
+        if duration > no_duration && now - start > duration {
+            print(
+                verbose,
+                format!(
+                    "Logs generation finished after {} seconds",
+                    (now - start).as_secs()
+                ),
+            );
+            break;
+        }
+
+        // Stop generation if configured limit reached
+        if total_logs != 0 && count_logs >= total_logs {
+            break;
+        }
+
+        // Generate logs from patterns
+        for pattern in &patterns {
+            let log = build_log(&pattern, &wordlist, &mut counter);
+            let mut saved_logs: u64 = 0;
+
+            match &fd {
+                Some(x) => saved_logs = save_log(&x, &log),
+                None => {}
+            }
+
+            if saved_logs == 1 {
+                current_logs += 1;
+                current_bytes += log.len() as u64;
+                count_logs += 1;
+                count_bytes += log.len() as u64;
+            }
+
+            if total_logs != 0 && count_logs >= total_logs {
+                break;
+            }
+        }
+    }
+    print(verbose, format!("Sent {} logs in total", count_logs));
+    print(verbose, format!("Sent {} bytes in total", count_bytes));
+}
+
+fn save_log(mut fd: &File, log: &String) -> u64 {
+    let write = fd.write_all(log.as_bytes());
+    match write {
+        Ok(_result) => {
```

Comment:
What is this rust wizardry? `_err` and `_result` are magic variable accessible when a function returns a `Result`?

Reply:
I believe this would be better explanation, than me trying to do that :D. [doc](https://doc.rust-lang.org/std/result/)